src/checkboxes/checkboxes.test.js

```diff
@@ -0,0 +1,178 @@
+/**
+ * @jest-environment ./lib/puppeteer/environment.js
+ */
+/* eslint-env jest */
+
+const cheerio = require('cheerio')
+
+const configPaths = require('../../config/paths.json')
+const PORT = configPaths.ports.test
+
+let browser
+let page
+let baseUrl = 'http://localhost:' + PORT
+
+const goToAndGetComponent = async (name, example) => {
+  const componentPath = `${baseUrl}/components/${name}/${example}/preview`
+  await page.goto(componentPath, { waitUntil: 'load' })
+  const html = await page.evaluate(() => document.body.innerHTML)
+  const $ = cheerio.load(html)
+  return $
+}
+
+const waitForHiddenSelector = async (selector) => {
+  return page.waitForSelector(selector, {
+    hidden: true,
+    timeout: 1000
+  })
+}
+
+const waitForVisibleSelector = async (selector) => {
+  return page.waitForSelector(selector, {
+    visible: true,
+    timeout: 1000
+  })
+}
+
+beforeEach(async () => {
+  browser = global.__BROWSER__
+  page = await browser.newPage()
+})
+
+afterEach(async () => {
+  await page.close()
+})
+
+describe('Checkboxes', () => {
+  describe('must be visible as static content if JavaScript is unavaliable or fails', () => {
+    it('with no aria attributes', async () => {
+      await page.setJavaScriptEnabled(false)
+
+      const $ = await goToAndGetComponent('checkboxes', 'with-conditional')
+      const $component = $('.govuk-checkboxes')
+
+      const hasAriaHidden = $component.find('[aria-hidden]').length
```

Comment:
The checkbox itself would never have `aria-hidden` right? That attribute would be on the conditional content? Should this be a separate test?

Reply:
`.find` will look for any child element with aria-hidden so this should catch any issues.

Reply:
Makes sense, though not sure that's clear. Can we be more specific and check the elements that we put the aria attributes on?

Reply:
Sorted